src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -109,19 +165,16 @@ public boolean equals(Object o) {
             return true;
         }
 
+        if (crc32 != that.crc32) {
+            return false;
+        }
+
         return Arrays.equals(uncompressed(), that.uncompressed());
     }
 
     @Override
     public int hashCode() {
-        if (hashCode == 0) {
-            int h = Arrays.hashCode(uncompressed());
-            if (h == 0) {
-                h = 1;
-            }
-            hashCode = h;
-        }
-        return hashCode;
+        return (int) crc32;
```

Comment:
it is not, I removed it. It is due to the fact that the crc32 was initially a long since Checksum.getValue returns a long but then I switched to an int since only the lower 32 bits are set